### PR TITLE
PARQUET-192: Fix map null encoding

### PR DESCRIPTION
**Pull request by rdblue**
This depends on PARQUET-191 for the correct schema representation.

### CODE REVIEW

File: parquet-avro/src/test/java/parquet/avro/TestReadWrite.java [hunk 1]
```diff
@@ -103,6 +104,62 @@ public void testEmptyMap() throws Exception {
...
+
+    File tmp = File.createTempFile(getClass().getSimpleName(), ".tmp");
+    tmp.deleteOnExit();
+    tmp.delete();
+    Path file = new Path(tmp.getPath());
+
+    AvroParquetWriter<GenericRecord> writer =
+        new AvroParquetWriter<GenericRecord>(file, schema);
+
+    // Write a record with an empty map.
```
Comment (tomwhite) [inline parquet-avro/src/test/java/parquet/avro/TestReadWrite.java]: Nit: should say empty (or null) _value_

**Comment by tomwhite** [comment]
This looks good. Looks like it overlaps with https://github.com/apache/incubator-parquet-mr/pull/126

**Comment by rdblue** [comment]
This PR is based on the fix for PARQUET-191, which is #126.

**Comment by rdblue** [comment]
I've fixed the problems and rebased on master now that PARQUET-191 (#126) is merged. Thanks for the review, Tom!

**Pull request closed without merging**